coffeecard/CoffeeCard.WebApi/Pages/Recover.cshtml.cs

```diff
@@ -25,9 +27,16 @@ public Recover(IAccountService accountService, ITokenService tokenService)
 
         public async Task<IActionResult> OnGet()
         {
-            IsTokenValid = await _tokenService.ValidateToken(Token);
+            try
+            {
+                IsTokenValid = await _tokenService.ValidateTokenIsUnusedAsync(Token);
 
-            if (IsTokenValid) return Page();
+                if (IsTokenValid) return Page();
+            }
+            catch (Exception e)
+            {
+                Log.Error("An unknown exception {ex} occured while validating token {token} for account recovery", e, Token);
+            }

```

Comment:
I think we must decide what pattern we go for when it comes to handling the invalid token and exception handling. 
We do it differently on different pages. Here we do if (no else), then a catch with logging and treat the invalid token and treat any exception the same when communicating to the user.
But e.g. in `VerifyEmail.cshtml.cs` we do have the else case, we catch the exception but do not log, and treat the invalid email error message and caught the exception error message uniquely.

On top of my head, I suggest that we apply a pattern closer to the last for **all** pages
1. Both if and else case
2. Catch Exception, log the exception
3. For failures, i.e. the _else_ case and the exception, the error message is set uniquely and describes something meaningful for the user

Reply:
This got refactored a bit. Now each page gives a Func to a PageUtil that then executes the func, and handles any exceptions that my occur from running the code. Let me know what you think of this approach